test/runtests.jl

```diff
@@ -1129,3 +1129,81 @@ end
         @test coef(glm_dense_alt) ≈ coef(glm_views_alt)
     end
 end
+
+@testset "PowerLink" begin
+    @testset "Functions related to PowerLink" begin
+        @test GLM.linkfun(IdentityLink(), 10) == GLM.linkfun(PowerLink(1), 10)
+        @test GLM.linkfun(SqrtLink(), 10) == GLM.linkfun(PowerLink(0.5), 10)
+        @test GLM.linkfun(LogLink(), 10) == GLM.linkfun(PowerLink(0), 10)
+        @test GLM.linkfun(InverseLink(), 10) ≈ GLM.linkfun(PowerLink(-1), 10)
+        @test GLM.linkfun(InverseSquareLink(), 10) ≈ GLM.linkfun(PowerLink(-2), 10)
+        @test isapprox(GLM.linkfun(PowerLink(1 / 3), 10), 2.154434690031884)
+
+        @test GLM.linkinv(IdentityLink(), 10) == GLM.linkinv(PowerLink(1), 10)
+        @test GLM.linkinv(SqrtLink(), 10) == GLM.linkinv(PowerLink(0.5), 10)
+        @test GLM.linkinv(LogLink(), 10) == GLM.linkinv(PowerLink(0), 10)
+        @test GLM.linkinv(InverseLink(), 10) ≈ GLM.linkinv(PowerLink(-1), 10)
+        @test GLM.linkinv(InverseSquareLink(), 10) ≈ GLM.linkinv(PowerLink(-2), 10)
+        @test isapprox(GLM.linkinv(PowerLink(1 / 3), 10), 1000.0)
+
+        @test GLM.mueta(IdentityLink(), 10) == GLM.mueta(PowerLink(1), 10)
+        @test GLM.mueta(SqrtLink(), 10) == GLM.mueta(PowerLink(0.5), 10)
+        @test GLM.mueta(LogLink(), 10) == GLM.mueta(PowerLink(0), 10)
+        @test GLM.mueta(InverseLink(), 10) ≈ GLM.mueta(PowerLink(-1), 10)
+        @test GLM.mueta(InverseSquareLink(), 10) == GLM.mueta(PowerLink(-2), 10)
+        @test isapprox(GLM.mueta(PowerLink(1 / 3), 10), 300.0)
+
+        @test PowerLink(1 / 3) == PowerLink(1 / 3)
+        @test isequal(PowerLink(1 / 3), PowerLink(1 / 3))
+        @test !isequal(PowerLink(1 / 3), PowerLink(0.33))
+        @test hash(PowerLink(1 / 3)) == hash(PowerLink(1 / 3))
+        @test hash(PowerLink(1 / 3)) != hash(PowerLink(0.33))
```

Comment:
`hash` isn't guaranteed to return different values for objects that are not equal.
```suggestion
```